### PR TITLE
fix(api): alloc and draw cursor window in nvim__redraw

### PR DESCRIPTION
**Pull request by luukvbaal**
Problem:  Unable to move cursor to recently opened window.
Solution: Make sure uninitialized window is drawn before trying to move
          the cursor to it.

Ref #28868

### CODE REVIEW

File: test/functional/api/vim_spec.lua [hunk 1]
```diff
@@ -5020,12 +5020,29 @@ describe('API', function()
...
+      width = 1,
+      height = 1,
+      row = 1,
+      col = 10,
+    })
+    api.nvim__redraw({ win = newwin, cursor = true })
+    screen:expect({
+      grid = [[
+                                                                    |
+        {1:~         ^                                                  }|
```
Comment (luukvbaal) [inline test/functional/api/vim_spec.lua:5041]: It's probably more user friendly/expected to just draw the window in this case, so flush implicitly.

File: src/nvim/api/vim.c [hunk 2]
```diff
@@ -2422,11 +2418,22 @@ void nvim__redraw(Dict(redraw) *opts, Error *err)
     }
   }
 
-  // Flush pending screen updates if "flush" or "clear" is true, or when
-  // redrawing a status component may have changed the grid dimensions.
+  win_T *cwin = win ? win : curwin;
+  // Allow moving cursor to recently opened window and make sure it is drawn(#28868).
```
Comment (zeertzjq) [inline src/nvim/api/vim.c:2422]:
```suggestion
  // Allow moving cursor to recently opened window and make sure it is drawn #28868.
```

File: test/functional/api/vim_spec.lua [hunk 3]
```diff
@@ -5020,12 +5020,29 @@ describe('API', function()
   it('nvim__redraw', function()
     local screen = Screen.new(60, 5)
     screen:attach()
-    local win = api.nvim_get_current_win()
     eq('at least one action required', pcall_err(api.nvim__redraw, {}))
     eq('at least one action required', pcall_err(api.nvim__redraw, { buf = 0 }))
     eq('at least one action required', pcall_err(api.nvim__redraw, { win = 0 }))
     eq("cannot use both 'buf' and 'win'", pcall_err(api.nvim__redraw, { buf = 0, win = 0 }))
+    local win = api.nvim_get_current_win()
+    -- Can move cursor to recently opened window and window is flushed(#28868)
```
Comment (zeertzjq) [inline test/functional/api/vim_spec.lua:5028]:
```suggestion
    -- Can move cursor to recently opened window and window is flushed #28868
```

**Comment by clason** [comment]
@luukvbaal rebase?

**Pull request merged**